internal/collector/scrape.go

```diff
@@ -202,11 +213,11 @@ func (c *Collector) scrapeLiquid(ctx context.Context, connection *core.LiquidCon
 	return resourceData, serializedMetrics, nil
 }
 
-func (c *Collector) writeResourceScrapeResult(dbDomain db.Domain, dbProject db.Project, task projectScrapeTask, resourceData liquid.ServiceUsageReport, serializedMetrics []byte) error {
+func (c *Collector) writeResourceScrapeResult(dbDomain db.Domain, dbProject db.Project, task projectScrapeTask, resourceData liquid.ServiceUsageReport, serializedMetrics []byte, serviceInfos map[db.ServiceType]liquid.ServiceInfo) error {
```

Comment:
As mentioned in previous notes, we should not need to take a structure that looks like it contains multiple ServiceInfo when we only have (and need) a single one:
```suggestion
func (c *Collector) writeResourceScrapeResult(dbDomain db.Domain, dbProject db.Project, task projectScrapeTask, resourceData liquid.ServiceUsageReport, serializedMetrics []byte, serviceInfo liquid.ServiceInfo) error {
```

Since `serviceInfos` is used to call into `InfoForResource`, maybe it makes sense to have `func InfoForResource` take `serviceInfo, resourceName` only instead of `serviceInfos, serviceType, resourceName`? This would make some callsites more tedious since we would have to call `InfoForService` first, but it would make this part easier to follow.